pkg/metallb/frr.go

```diff
@@ -0,0 +1,664 @@
+package metallb
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	frrtypes "github.com/openshift-kni/eco-goinfra/pkg/schemes/metallb/frrtypes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FrrConfigurationBuilder provides struct for the FrrConfiguration object containing connection to
+// the cluster and the FrrConfiguration definitions.
+type FrrConfigurationBuilder struct {
+	Definition *frrtypes.FRRConfiguration
+	Object     *frrtypes.FRRConfiguration
+	apiClient  *clients.Settings
+	errorMsg   string
+}
+
+// NewFrrConfigurationBuilder creates a new instance of FRRConfiguration.
+func NewFrrConfigurationBuilder(
+	apiClient *clients.Settings, name, nsname string) *FrrConfigurationBuilder {
+	glog.V(100).Infof(
+		"Initializing new Frrconfiguration structure with the following params: %s, %s",
+		name, nsname)
+
+	builder := FrrConfigurationBuilder{
+		apiClient: apiClient,
+		Definition: &frrtypes.FRRConfiguration{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       FrrConfigurationKind,
+				APIVersion: fmt.Sprintf("%s/%s", FRRAPIGroup, APIVersion),
+			},
```

Comment:
TypeMeta doesn't need to be included (except for manually creating policies with OCM)

Reply:
removed

Reply:
Test cases fail with 
			TypeMeta: metav1.TypeMeta{
				Kind:       FrrConfigurationKind,
				APIVersion: fmt.Sprintf("%s/%s", FRRAPIGroup, APIVersion),
			},

Message: "the object provided is unrecognized (must be of type FRRConfiguration): Object 'Kind' is missing in '{\"metadata\":

Returned to file

Reply:
@gkopels I think you need to use the runtime client (with scheme added) for it to work

Reply:
Yes worked after I added the  runtime client